### DISCUSSION TITLE
response_latency_ms_bucket

### DISCUSSION THREAD

**Discussion by ghost** [linkerd/linkerd2#8393] [Help] [answered]
Can you please explain the meaning of metric `response_latency_ms_bucket`. Is that just a name change for `response_latency_ms`? 
```
$ linkerd diagnostics proxy-metrics -n linkerd deploy/linkerd-controller | grep -B 2 -m 1 response_latency_ms_bucket
# HELP response_latency_ms Elapsed times between a request's headers being received and its response stream completing
# TYPE response_latency_ms histogram
response_latency_ms_bucket{direction="inbound",target_addr="0.0.0.0:4191",tls="true",client_id="prometheus.linkerd-viz.serviceaccount.identity.linkerd.cluster.local",status_code="200",le="1"} 548421
```

> response_latency_ms: A histogram of response latencies. This measurement reflects the [time-to-first-byte](https://en.wikipedia.org/wiki/Time_to_first_byte) (TTFB) by recording the elapsed time between the proxy processing a request’s headers and the first data frame of the response. If a response does not include any data, the end-of-stream event is used. The TTFB measurement is used so that Linkerd accurately reflects application behavior when a server provides response headers immediately but is slow to begin serving the response body.

Thank you.

**Comment by olix0r** [comment, marked as answer]
This is a [Prometheus histogram](https://prometheus.io/docs/concepts/metric_types/#histogram):

> A histogram with a base metric name of <basename> exposes multiple time series during a scrape:
>
> * cumulative counters for the observation buckets, exposed as `<basename>_bucket{le="<upper inclusive bound>"}`
> * the total sum of all observed values, exposed as `<basename>_sum`
> * the count of events that have been observed, exposed as <basename>_count (identical to `<basename>_bucket{le="+Inf"}` above)
